uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/LeaveRequestRights.php

```diff
@@ -151,4 +152,25 @@ private static function getLeaveRequestStatuses() {
 
     return self::$leaveStatuses;
   }
+
+  /**
+   * Checks whether the current user can cancel the TOIL Request with
+   * past dates or not.
+   *
+   * @param int $contactID
+   *   The contactID of the leave request
+   * @param int $absenceTypeID
+   *   The absence Type ID of the leave request
+   *
+   * @return bool
+   */
+  public function canCancelToilWithPastDates($contactID, $absenceTypeID) {
```

Comment:
Would it be possible to move all the logic for `canCancel` into a single `canCancel` method like:

```
public function canCancel($leaveRequestId, $contactId)
```

It might be more readable and avoid duplication of code

Reply:
Not sure I understand what you mean here. There is no method named `canCancel` presently. And I didn't want to overcomplicate things. If eventually the need arises for such a method, this method `canCancelToilWithPastDates` can be made private and can be used in the `canCancel` method.

Reply:
Fair enough, my reasoning was just that it might be more readable to have `canCancel` which would check `canCancelToilWithPastDates` if it's in the past, but it might be going a step too far. I guess if the validation gets more complicated in the future we could refactor it to something like that then.